src/test/steps/common.ts

```diff
@@ -268,42 +271,43 @@ export const iGetTheCaseApi = (testUser: UserDetails): CaseApi => {
 };
 
 export const iSetTheUsersCaseTo = async (userCaseObj: Partial<BrowserCase>): Promise<void> =>
-  executeUserCaseScript(userCaseObj, RELATIONSHIP_DATE_URL, WHERE_YOUR_LIVES_ARE_BASED_URL);
+  executeUserCaseScript(userCaseObj, WHERE_YOUR_LIVES_ARE_BASED_URL);
 
 export const iSetApp2UsersCaseTo = async (userCaseObj: Partial<BrowserCase>): Promise<void> =>
-  executeUserCaseScript(userCaseObj, APPLICANT_2 + HAS_RELATIONSHIP_BROKEN_URL, APPLICANT_2 + YOUR_NAME);
+  executeUserCaseScript(userCaseObj, APPLICANT_2 + YOUR_NAME);
 
 export const iSetRespondentUsersCaseTo = async (userCaseObj: Partial<BrowserCase>): Promise<void> =>
-  executeUserCaseScript(
-    userCaseObj,
-    RESPONDENT + HOW_DO_YOU_WANT_TO_RESPOND,
-    RESPONDENT + LEGAL_JURISDICTION_OF_THE_COURTS
-  );
-
-const executeUserCaseScript = (userCaseObj, requestPageLink: string, redirectPageLink: string) =>
-  I.executeScript(
-    async ([userCase, requestUrl, redirectUrl]) => {
-      const mainForm = document.getElementById('main-form') as HTMLFormElement;
-      const formData = mainForm ? new FormData(mainForm) : new FormData();
-      for (const [key, value] of Object.entries(userCase)) {
-        formData.set(key, value as string);
-      }
+  executeUserCaseScript(userCaseObj, RESPONDENT + LEGAL_JURISDICTION_OF_THE_COURTS);
+
+const executeUserCaseScript = async (data, redirectPageLink: string) => {
+  await I.grabCurrentUrl();
+
+  const user = testConfig.GetCurrentUser();
+  const testUser = await iGetTheTestUser(user);
+  const api = iGetTheCaseApi(testUser);
+  const userCase = await api.getOrCreateCase(DivorceOrDissolution.DIVORCE, testUser);
+
+  data.applicant2MiddleNames = data.state || userCase.state;
 
-      const request = {
-        method: 'POST',
-        body: new URLSearchParams(formData as unknown as Record<string, string>),
-      };
+  const connections = addConnection(data);
 
-      await fetch(requestUrl, request);
-      await new Promise(resolve => setTimeout(resolve, 1500));
-      await fetch(redirectUrl, request);
-    },
-    [userCaseObj, requestPageLink, redirectPageLink]
-  );
+  // don't set as applicant 2 as they don't have permission
+  data.connections = connections.length > 0 ? connections : undefined;
```

Comment:
would this still not set it for app2? i guess app2 will only use jointApp2 fixture and the data isnt enough?

Reply:
yeah, it's not very robust but I couldn't think of an easy way to work out whether it's app two